File: benchmarks/python/string_concat.py
import sys

n = int(sys.argv[1]) if len(sys.argv) > 1 else 20_000
acc = ""

for i in range(n):
    if i % 7 == 0:
        acc += "x"
    elif i % 3 == 0:
        acc += "ab"
    else:
        acc += "1"

print(len(acc))

File: benchmarks/php/string_concat.php
<?php
$n = isset($argv[1]) ? (int) $argv[1] : 20000;
$acc = "";

for ($i = 0; $i < $n; $i++) {
    if ($i % 7 === 0) {
        $acc .= "x";
    } elseif ($i % 3 === 0) {
        $acc .= "ab";
    } else {
        $acc .= "1";
    }
}

echo strlen($acc), PHP_EOL;
